Sources/SwiftSyntaxMacros/Syntax+LexicalContext.swift

```diff
@@ -56,6 +56,14 @@ extension SyntaxProtocol {
       patternBinding.accessorBlock = nil
       patternBinding.initializer = nil
       return Syntax(patternBinding)
+        
+    // Freestanding macros have their arguments and trailing closures removed.
+    case var freestandingMacro as FreestandingMacroExpansionSyntax:
+      freestandingMacro = freestandingMacro.detached
+      freestandingMacro.arguments = LabeledExprListSyntax()
+      freestandingMacro.trailingClosure = nil
+      freestandingMacro.additionalTrailingClosures = MultipleTrailingClosureElementListSyntax()
+      return Syntax(freestandingMacro) as Syntax
```

Comment:
Shouldn’t we also remove `leftParen` and `rightParen`? What about the `genericArgumentClause`? Should that stay or go? And lastly `attributes` and `modifiers` for `MacroExpansionDeclSyntax`? I would think that all of them should be removed. 

@DougGregor what do you think?

Reply:
I was following the lead of the other cases -- functions and types keep their parens and their attributes and modifiers.

Reply:
I wasn't sure about the generic argument clause, since a freestanding macro is a bit more like a function call than a declaration. Happy to remove it.

Reply:
So I, personally, would say that we can just return `freestandingMacro.detached` here

Reply:
Ah -- I didn't understand the purpose of removing that information, but that makes sense. I'll fix.